frontend/packages/console-shared/src/hooks/usePolls.ts

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import { PollsDataMap, UsePolls } from '@console/dynamic-plugin-sdk/src/api/internal-types';
+
+// Modified from Dan Abramov's blog post about using React hooks for polling
+// https://overreacted.io/making-setinterval-declarative-with-react-hooks/
+const usePolls: UsePolls = (pollsDataMap) => {
```

Comment:
maybe we will want to merge logic of this hook to `useURLPolls`. I think we want to abort all pending requests in case of retriggering `useEffect` in which case `callback` has to be a cancellable Promise and then I think we dont really need to split `usePolls` and `useURLPolls`